tool/canonical/scorecard.dart

```diff
@@ -0,0 +1,403 @@
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/src/lint/config.dart'; // ignore: implementation_imports
+import 'package:analyzer/src/lint/registry.dart'; // ignore: implementation_imports
+import 'package:github/github.dart';
+import 'package:http/http.dart' as http;
+import 'package:linter/src/analyzer.dart';
+import 'package:linter/src/rules.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as path;
+
+import '../parse.dart';
+
+/// Issue triage: go through all issues and tag bugs
+/// Consider new tags for canonical-blocking work
+/// False positive vs. false negative?
+
+void main() async {
+  var scorecard = await ScoreCard.calculate();
+  var details = [
+    Detail.rule,
+    Detail.fix,
+    // Detail.status,
+    Detail.score,
+    Detail.recommend,
+    Detail.bugs,
+  ];
+
+  var sorter = (LintScore s1, LintScore s2) {
+    var base = _compareRuleSets(s1.ruleSets, s2.ruleSets) * 1000;
+    return s1.name.compareTo(s2.name) + base;
+  };
+
+  print(scorecard.asMarkdown(details, sorter: sorter));
+
+  var footer = buildFooter(scorecard, details);
+  print(footer);
+}
+
+const bulb = '💡';
+const checkMark = '✅';
+
+Iterable<LintRule> _registeredLints;
+
+Iterable<String> get registeredLintNames => registeredLints.map((r) => r.name);
+
+Iterable<LintRule> get registeredLints {
+  if (_registeredLints == null) {
+    registerLintRules();
+    _registeredLints = Registry.ruleRegistry.toList()
+      ..sort((l1, l2) => l1.name.compareTo(l2.name));
+  }
+  return _registeredLints;
+}
+
+StringBuffer buildFooter(ScoreCard scorecard, List<Detail> details) {
+  var scoreLintCount = 0;
+  var scoreFixCount = 0;
+
+  var recommendLintCount = 0;
+  var recommendFixCount = 0;
+
+  var needsBulkFix = <String>[];
+
+  for (var score in scorecard.scores) {
+    for (var ruleSet in score.ruleSets) {
+      if (ruleSet == 'score') {
+        ++scoreLintCount;
+        if (score.hasFix) {
+          ++scoreFixCount;
+        }
+      }
+      if (ruleSet == 'recommend') {
+        ++recommendLintCount;
+        if (score.hasFix) {
+          ++recommendFixCount;
+        }
+      }
+      if (score.hasFix && !score.hasBulkFix) {
+        needsBulkFix.add(score.name);
+      }
+    }
+  }
+
+  var footer = StringBuffer('\n${scorecard.lintCount} lints: ');
+  footer.write('$scoreLintCount score [$scoreFixCount fixes], ');
+  footer.write('rec $recommendLintCount [$recommendFixCount fixes]');
+  if (needsBulkFix.isNotEmpty) {
+    footer.writeln('\n\nTODO: add bulk fixes for');
+    for (var lint in needsBulkFix) {
+      footer.writeln('  - [ ] `$lint`');
+    }
+  }
+
+  return footer;
+}
+
+int _compareRuleSets(List<String> s1, List<String> s2) {
+  if (s1.contains('score')) {
+    return s2.contains('score') ? 0 : -1;
+  }
+  if (s2.contains('score')) {
+    return 1;
+  }
+  return 0;
+}
+
+//bool _isBug(Issue issue) => issue.labels.map((l) => l.name).contains('bug');
```

Comment:
Remove commented out code (both here and below)?

Reply:
I'm going to use it shortly; just haven't wired it up yet.